internal/set.go

```diff
@@ -0,0 +1,67 @@
+package internal
```

Comment:
1. Please look for any good poackage for set datatype
2. If we are keeping this, please move it to a proper package